tests/terraform/checks/resource/aws/test_IAMAdminPolicyDocument.py

```diff
@@ -13,7 +13,7 @@ def test_success(self):
                          "Action": [\n        "ec2:Describe*"\n      ],\n      \
                          "Effect": "Allow",\n     \
                           "Resource": "*"\n    }\n  ]\n}']}
-        scan_result = check.scan_entity_conf(conf=resource_conf)
+        scan_result = check.scan_entity_conf(conf=resource_conf, entity_type='aws_iam_policy')
```

Comment:
why is entity_type addition required?

Reply:
As mentioned in #45 the wildcard feature could be used for a naming convention. In the case of aws with terraform this is usually name but e.g. for S3 buckets this is bucket. `entity_type` is a way to catch such edge cases. Checking if `conf` contains the value is possible but I can't guarantee that this not lead to false positives in some cases. Giving a ground truth about the type gives such guarantee.

Do you have another solution?

Reply:
@angatha If I understand correctly, this change is not required. Right?
If that's so - let's revert changes to the old tests (`test_IAMAdminPolicyDocument.py` & `test_IAMStarActionPolicyDocument.py`) please

Reply:
@schosterbarak sorry, when I first answered to your question I hade something else in mind.

@nimrodkor I would say it is required. Let me explain, why entity_type on `checkov.common.checks.base_check.BaseCheck.scan_entity_conf(...)` does not have a default value. This then implies, why the test, that called this directly needs to be adjusted.

When writing a check, that uses `entity_type`, you don't want to check if it is not None, because how should a check, that requires this information, handle it? It is nothing that depends on the files checkov is scanning but entirely on the implementation of checkov. And this implementation can guarantee that when running a check by calling `scan_entity_conf` from `checkov.common.checks.base_check.BaseCheck.run(...)` is not None (we can anotate the type as `str` and not `typing.Optional[str]`).

At this point, the check implementation expects a `str` and therefore the check code should pass the correct `entity_type`. (Even if the check doesn't use it.)

Reply:
When patching the function, I set a default value. This (unintentionally) hides other places, where this would occur. But in this cases, the test calls the function on the check directly e.g. `tests.terraform.checks.resource.aws.test_IAMPolicyAttachedToGroupOrRoles`. There you can argument, that the function doesn't define it on the other hand, the new signature specifies it.